ompy/error_finder.py

```diff
@@ -0,0 +1,433 @@
+
+        Raises:
+            NotImplementedError if algorithm isn't recognized.
+
+        """
+
+        if not (algorithm.lower() == 'lin' or algorithm.lower() == 'log'):
+            raise NotImplementedError(
+                f"Algorithm '{algorithm}' not implemented")
+
+        self.algorithm = algorithm.lower()
+        self.options = {} if options is None else options
+        self.options.setdefault("return_inferencedata", False)
+        self.seed = seed
+        self.testvals = {'σ_D': 0.25, 'σ_F': 0.25, 'σ_α': 0.05,
+                         'σ_A': 0.25, 'σ_B': 0.25, 'A': 1.0, 'B': 1.0}
+
+    def __call__(self, *args) -> Union[Tuple[Vector, Vector], Any]:
+        """ Wrapper for evaluate """
+        return self.evaluate(*args)
+
+    def evaluate(self, nlds: List[Vector],
+                 gsfs: List[Vector],
+                 median: Optional[bool] = False,
+                 full: Optional[bool] = False
+                 ) -> Union[Tuple[Vector, Vector], Any]:
+        """ Evaluate the model and find the relative errors
+        of the NLDs and γSFs passed to the function.
+
+        Args:
+            nlds (List[Vector]): List of the NLDs in an ensemble
+            gsfs (List[Vector]): List of the γSFs in an ensemble
+            median (bool, optional): If the mean of the relative error should
+                be used rather than the mean. Default is False.
+            full (bool, optional): If the sample trace should
+                be returned or not.
+        Returns:
+            The relative error of the nuclear level density and
+            gamma strength function.
+            Optionally returns the full sample trace if `trace` is true.
+
+        Raises:
+            ValueError: If length of `nlds` doesn't match the length of `gsfs`.
+            pyMC3 may raise errors. Please report if such errors occurs.
+        """
+        algo = None
+        if self.algorithm == 'lin':
+            algo = self.linear
+        elif self.algorithm == 'log':
+            algo = self.logarithm
+        else:
+            raise NotImplementedError(
+                f"Algorithm '{algorithm}' not implemented")
+
+        return algo(*self.condition_data(nlds, gsfs),
+                    median=median, full=full)
+
+    def logarithm(self, E_nld: ndarray, E_gsf: ndarray,
+                  nld_obs: ndarray, gsf_obs: ndarray,
+                  coef_nld: ndarray, coef_gsf: ndarray,
+                  vals_nld: ndarray, vals_gsf: ndarray,
+                  median: Optional[bool] = False, full: Optional[bool] = False
+                  ) -> Union[Tuple[Vector, Vector], Any]:
+        """ Calculate the relative errors of the NLD and γSF points using a
+        logarithmic model.
+
+        Args:
+            E_nld (ndarray): Energy points of the NLD
+            E_gsf (ndarray): Energy points of the γSF
+            nld_obs (ndarray): The observation tensor of the NLD
+            gsf_obs (ndarray): The observation tensor of the γSF
```

Comment:
Where and everywhere else: Would it be better to say _ref and the *reference* tensor, instead of *observed*?